x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/StepsFactory.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.indexlifecycle;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.logging.ESLoggerFactory;
+import org.elasticsearch.xpack.core.indexlifecycle.Step.StepKey;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.function.LongSupplier;
+
+public class StepsFactory {
```

Comment:
Can you move this class, all the `ActionStepCreator`s and all the `Step`s into `:x-pack:index-lifecycle`?

Reply:
yeah. as noted in my description, I was purposefully planning on moving much of it all later anyways. but I can move the new stuff

Reply:
StepsFactory references PhaseAfterStep. Until that is moved, StepsFactory actually has to stay in core